src/main/java/io/github/guacsec/trustifyda/integration/providers/ProviderResponseHandler.java

```diff
@@ -447,7 +471,10 @@ private SourceSummary buildSummary(
   }
 
   private void incrementCounter(PackageItem item, VulnerabilityCounter counter, boolean isDirect) {
-    if (item != null && !item.issues().isEmpty()) {
+    if (item == null) {
+      return;
+    }
+    if (!item.issues().isEmpty()) {
```

Comment:
The old implementation can throw a NPE.
If item.issues() returns null, then line this line would throw a NullPointerException before we ever reach the null check at below.

Reply:
Fixed, thanks